src/CMakeLists.txt

```diff
@@ -1,8 +1,10 @@
 # Source files
-file(GLOB_RECURSE SOURCES_SC2LADDERSERVER "*.cpp" "*.h")
+file(GLOB SOURCES_SC2LADDERSERVER "Sc2LadderServer/*.cpp" "Sc2LadderServer/*.h")
+file(GLOB SOURCES_SC2LADDERCORE "core/*.cpp" "core/*.h")
```

Comment:
Like so?
```
src/Sc2LadderServer/CMakeLists.txt
src/core/CMakeLists.txt
```

Would you keep `src/CMakeLists.txt` or just link straight from the root `CMakeLists.txt`?

Reply:
A common approach is to keep CMakeLists in almost every nested directory especially when one compiles additional libraries.

Thus it could be:
```cmake
src/Sc2LadderServer/CMakeLists.txt # Compile the ladder server binary.
src/tests/CMakeLists.txt #include all the nested tests directories (unit-tests, debug bot, functional tests).
src/tests/DebugBot/CMakeLists.txt # Compile the debug bot.
src/core/CMakeLists.txt # Compile the common library.
src/CMakeLists.txt #Check dependencies if needed, handle global variables, apply project-wide settings.
```

Reply:
I looked into doing this and it became a bit of a job in itself.
I think I will hold off for now and look to do the nested CMakeLists a bit later, once I have my head around CMake a bit better.
The way I've set it up is how the s2client-api is setup, so hopefully it will do for the time being.